### PR TITLE
Use AggregatingTestRunner for non-trivial cases

### LINKED ISSUE #730: NUnit project file settings are ignored
The way TestPackages deal with settings means that settings always propagate down to sub packages, but never up. At launch, all console line settings are set on the top level package, and are propagated down to the next package, which could be an .nunit project.

This subpackage later (at expand time) loads its settings, such as process model and domain model, from the project file. These settings are never propagated back up to the top level TestPackage, however, so they never take proper effect.

This was already a problem in 3.10, but it still managed to work because NUnit would (incorrectly, I think) create an AggregateTestRunner for the single .nunit project, which would then in turn instantiate the correct runner (say a multi process version).

Since the aggregate test runner has been removed, this is no longer the case and NUnit will always use a single process runner even when the .nunit project specifies otherwise. In my case, this causes all my tests to fail.

So, this is a regression since 3.10, but in my opinion the settings handling was already broken earlier.

The simplest fix is this: Whenever GetSetting is called and we have exactly one sub package, forward the call to the sub package instead of handling it ourselves. I've created a pull request for this fix.

If you'd prefer a different fix that's fine, but it will leave us in the odd situation of taking action based on only console settings in several places.

### PR DESCRIPTION
**Pull request by cyanite**
This means cases involving at least one project file.

Fixes #730

### CODE REVIEW

**Comment by cyanite** [comment]
See discussion in #730

**Comment by CharliePoole** [comment]
@cyanite Your arguments for putting in this fix also work as arguments for it's being a critical bug - which we have always defined as one requiring an immediate pre-release.

@ChrisMaddock classified it as merely high priority, which would mean fix right away but don't release. That said, I could be reading too much into the classification. For me, it's good to have that classification right up front, because I would review a critical bug differently from any other. (FWIW I feel personally that this one is critical, because it broke a key feature.

That said, if it's not handled as critical, I would lean toward taking another week or come up with a fix that would survive new features that might come along. For example... nested projects, pre-expanded saved packages, etc. To me, that means actually checking to see if there are any projects involved, at any depth in the tree, using the ProjectService to recognize projects. That's not terribly hard to do, in fact, so i would lean to tossing the logic that uses subpackage count at all... btw, I think I may have been the one to come up with that logic, but things have moved on since then in other parts of the code, just not at the particular point where we are now looking.

@ChrisMaddock FWIW, in the long term, I think we probably want a bottom-up selection of runners, rather than top-down. That is, look at the lowest level subpackages first. Then, if a package has two subpackages with different settings, select the appropriate combining runner for those two packages. That would be a complete rewrite however, probably for NUnit 4 or later.

Bottom line, let's decide if this is critical. If a fix is going to be released in a matter of days, then the one @cyanite has done is perfect. But if it's just going to sit in master for a few months, I'd like to see a better one. I recognize that anyone can use the dev build if they need this fix, but I don't think that happens for a lot of organizations.

Finally, if we do this as a critical fix, can we leave the issue (or create a new one) so the fix gets fixed at a later time?

**Comment by ChrisMaddock** [comment]
Thanks for digging into this! Will try and take a proper look at the weekend. 🙂

**Comment by ChrisMaddock** [comment]
Hmm...this is irritating. Turns out we actually picked up on this change causing a potential problem when it was implemented - we just missed what the actual effect would be.

https://github.com/nunit/nunit-console/pull/622/files#r287597965


Thanks for looking into this, @cyanite. Merging this as a temporary fix.

At some point, we need to rewrite this area fully, it's proving unmaintainable. I'd like to do that as part of v4 - and remove domain isolation at the same time.
